src/views/Join/index.js

```diff
@@ -1,30 +1,158 @@
+      },
+      () => {
+        if (this.state.partyId) this.getParty();
+      }
+    );
+  }
+
+  onChangeText = partyId => {
+    this.setState({
+      partyId,
+    });
+  };
+
+  getParty = async () => {
+    const { partyId } = this.state;
+    const party = await this.db.getPartyById(partyId);
+
+    if (!party) {
+      alert(`No party with id ${partyId}`);
+      return;
+    }
+
+    this.props.navigation.navigate('Lobby', { party });
+  };
+
+  openCamera = async () => {
+    const { status } = await Permissions.askAsync(Permissions.CAMERA);
+
+    if (status !== 'granted') {
+      alert(
+        'Camera permissions have been denied. Go into your settings and allow camera access.'
+      );
+    }
+
+    this.setState({
+      hasCameraPermission: status === 'granted',
+      openCamera: status === 'granted',
+    });
+  };
+
+  onBarCodeRead = qr => {
+    if (this.state.invalidCameraState) {
+      return;
+    }
+
+    const partyId = Linking.parse(qr.data).queryParams.id;
+
+    if (!partyId) {
+      this.setState({ invalidCameraState: true });
+      Alert.alert('Invalid QR', `QR: ${qr.data} is not a valid party`, [
+        {
+          text: 'OK',
+          onPress: () => this.setState({ invalidCameraState: false }),
+        },
+      ]);
+      return;
+    }
+
+    this.setState(
+      {
+        partyId,
+        openCamera: false,
+      },
+      () => this.getParty()
+    );
+  };
+
   render() {
+    if (this.state.openCamera) {
+      return (
+        <Camera
+          type={Camera.Constants.Type.back}
+          style={{ flex: 1 }}
+          onBarCodeRead={this.onBarCodeRead}
+        >
+          <View>
+            <TouchableOpacity
+              onPress={() => {
+                this.setState({
+                  type:
+                    this.state.type === Camera.Constants.Type.back
+                      ? Camera.Constants.Type.front
+                      : Camera.Constants.Type.back,
+                });
+              }}
+            />
+          </View>
+        </Camera>
+      );
+    }
+
     return (
       <Container>
         <Content scrollEnabled={false} contentContainerStyle={styles.content}>
-          <Item floatingLabel >
-            <Label> Party Code </Label>
-            <Input />
+          <Item floatingLabel>
+            <Label>Party Code</Label>
+            <Input
+              value={this.state.partyId}
+              onChangeText={this.onChangeText}
+              required
+            />
           </Item>
           <Button
             iconRight
             block
             bordered
             info
             style={styles.buttonBlock}
-            onPress={() => this.props.navigation.navigate('Lobby')}
+            onPress={this.openCamera}
           >
-            <Text> Scan </Text>
-            <Icon name='md-qr-scanner' />
+            <Text>Scan</Text>
+            <Icon name="md-qr-scanner" />
```

Comment:
Should we update our prettier config file to use single quotes?

Reply:
Airbnb style-guide uses single quotes for strings but double quotes for HTML/JSX attributes.  I think it's nice to have that small difference but we can change it.